extensions/amp-accordion/1.0/accordion.js

```diff
@@ -88,15 +89,26 @@ function AccordionWithRef(
           expandSingleSection
         );
       });
-      return () => setExpandedMap((expandedMap) => omit(expandedMap, id));
+      eventMap.current = {...eventMap.current, [id]: onExpandStateChange};
+      return () => {
+        setExpandedMap((expandedMap) => omit(expandedMap, id));
+        eventMap.current = omit(/** @type {!Object} */ (eventMap.current), id);
+      };
     },
     [expandSingleSection]
   );
 
   const toggleExpanded = useCallback(
     (id) => {
+      let newValue;
       setExpandedMap((expandedMap) => {
-        const newValue = !expandedMap[id];
+        newValue = !expandedMap[id];
```

Comment:
Ok. So from what I understood about expand-single-selection mode, this is not quite correct. What should happen here, I think, is:

1. Call `setExpanded(...)` just as below. Save it as `newExpandedMap`.
2. Compare the delta between the `expandedMap` and `newExpandedMap`. Some IDs will flip from `true` to `false` and vice versa.
3. For each delta find and fire the `onExpandStateChange` callback with the new value.

Reply:
Perfect, this was the missing piece for `expand-single-selection` mode, thanks!  See the latest, I calculate a list of changed sections within the `setExpandedMap` callback.  Then use a single Promise to fire all events (also within the callback).